packages/accounts-app/src/KeyringContext/KeyringContext.tsx

```diff
@@ -2,13 +2,23 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
+import accountObservable from '@polkadot/ui-keyring/observable/accounts';
 import keyring from '@polkadot/ui-keyring';
 import { KeyringOptions } from '@polkadot/ui-keyring/types';
-import { ApiContext } from '@substrate/context';
-import React, { useContext, useEffect } from 'react';
+import { System } from '@substrate/context';
+import React, { useEffect, useState } from 'react';
 
 interface KeyringContext {
+  allAccounts?: string[];
   keyring: typeof keyring;
+  keyringReady: boolean;
+}
+
+interface KeyringContextProps {
+  api?: any,
+  children: React.ReactNode,
+  isReady?: boolean,
+  system?: System
```

Comment:
Since KeyringContext is inside ApiContext, you could remove all these from props, and use `const { api, system, isReady } = useContext(ApiContext)`

Reply:
ah, so I forgot to mention this, but for some reason all of these are `undefined` when doing `const { api, system, isReady } = useContext(ApiContext)`... Similar to the redirect issue, it seems the component rerender on context value change is not happening as expect so I had to workaround like this.

Reply:
Ah ok, so let's keep it like this for now